apps/linode-marketplace-drupal/roles/drupal/tasks/main.yml

```diff
@@ -0,0 +1,88 @@
+---
+# Main Drupal installation tasks
+- name: Import MariaDB tasks
+  import_tasks: mariadb.yml
+
+- name: Import PHP tasks
+  import_tasks: php.yml
+
+- name: Import Apache tasks
+  import_tasks: apache.yml
+
+# Install Composer first
+- name: Install Composer
+  shell: |
+    curl -sS https://getcomposer.org/installer | php
+    mv composer.phar /usr/local/bin/composer
+  args:
+    creates: /usr/local/bin/composer
+
+# Create Composer cache directory with correct permissions
+- name: Create Composer cache directory
+  file:
+    path: /var/www/.cache/composer
+    state: directory
+    owner: www-data
+    group: www-data
+    mode: '0755'
+    recurse: yes
+
+# Ensure Composer cache directory and subdirectories are writable
+- name: Ensure Composer cache directory is writable
+  file:
+    path: /var/www/.cache/composer/repo/https---repo.packagist.org
+    state: directory
+    owner: www-data
+    group: www-data
+    mode: '0755'
+    recurse: yes
+
+# Create target directory with correct permissions
+- name: Create Drupal directory
+  file:
+    path: /var/www/drupal
+    state: directory
+    owner: www-data
+    group: www-data
+    mode: '0755'
+
+# Install Drupal using Composer
+- name: Create Drupal project using Composer
+  composer:
+    command: create-project
+    arguments: "drupal/recommended-project {{ drupal_docroot | default('/var/www/drupal') }} --stability stable --no-interaction"
+    working_dir: /var/www/drupal
+  become: yes
+  become_user: www-data
+
+- name: Create files directory
+  file:
+    path: /var/www/drupal/web/sites/default/files
+    state: directory
+    mode: '0755'
+    owner: www-data
+    group: www-data
+
+- name: Copy default settings file
+  copy:
+    src: /var/www/drupal/web/sites/default/default.settings.php
+    dest: /var/www/drupal/web/sites/default/settings.php
+    remote_src: yes
+    owner: www-data
+    group: www-data
+    mode: '0644'
+
+- name: Append trusted host patterns to settings.php
+  blockinfile:
+    path: /var/www/drupal/web/sites/default/settings.php
+    block: "{{ lookup('template', 'settings-append.php.j2') }}"
+    marker: "// {mark} ANSIBLE MANAGED BLOCK - TRUSTED HOST PATTERNS"
```

Comment:
I think it would make more sense to stick with `blockinfile` no? 
The settings I'm adding look like this:
```
// Trusted host patterns
$settings['trusted_host_patterns'] = [
  '^localhost$',
  '^127\.0\.0\.1$',
  '^\[::1\]$',
  '^{{ _domain | replace('.', '\.') }}$',
];
```
blockinfile seems to make more sense since:
- It's easier to read 
- It's easier for the end user to edit/add/remove lines if they want, when the patterns are broken out on separate lines

My two cents.